### PR TITLE
add metadata to new and upload cmds

### PR DESCRIPTION
**Pull request by daveads**
/claim #13

* Added --metadata to new.go and upload.go 
* created tests...

### CODE REVIEW

**Review by ezekg — CHANGES_REQUESTED** [review]
A few changes.

File: cmd/upload.go [hunk 1]
```diff
@@ -265,6 +268,13 @@ func uploadRun(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
+	var metadata map[string]interface{}
```
Comment (ezekg) [inline cmd/upload.go:271]: Could you also move this up before the `release` var, to save an API request in cases where the metadata is invalid?

File: cmd/upload.go [hunk 2]
```diff
@@ -380,21 +391,10 @@ func calculateSignature(encSigningKey string, file *os.File) (string, error) {
 		if err != nil {
 			return "", err
 		}
-	case "ed25519":
```
Comment (ezekg) [inline cmd/upload.go:383]: Why did you remove this? Please readd.

File: cmd/new.go [hunk 3]
```diff
@@ -57,9 +60,7 @@ func init() {
 	draftCmd.Flags().StringVar(&draftOpts.Package, "package", "", "package identifier for the release")
 	draftCmd.Flags().BoolVar(&draftOpts.NoAutoUpgrade, "no-auto-upgrade", false, "disable automatic upgrade checks [$KEYGEN_NO_AUTO_UPGRADE=1]")
 	draftCmd.Flags().StringSliceVar(&draftOpts.Entitlements, "entitlements", []string{}, "comma seperated list of entitlement constraints (e.g. --entitlements <id>,<id>,...)")
-
-	// TODO(ezekg) Prompt multi-line description input from stdin if "--"?
```
Comment (ezekg) [inline cmd/new.go:61]: Please keep this description TODO.

File: cmd/upload.go (hunk 1 above)
Comment (daveads) [reply cmd/upload.go:271]: okay

File: cmd/new.go [hunk 4]
```diff
@@ -62,6 +62,9 @@ func init() {
 	draftCmd.Flags().StringSliceVar(&draftOpts.Entitlements, "entitlements", []string{}, "comma seperated list of entitlement constraints (e.g. --entitlements <id>,<id>,...)")
 	draftCmd.Flags().StringVar(&draftOpts.Metadata, "metadata", "", "JSON string of metadata key-value pairs")
 
+	// TODO(ezekg) Prompt multi-line description input from stdin if "--"?
+	// TODO(ezekg) Add metadata Flags
```
Comment (ezekg) [inline cmd/new.go:66]: You can drop this TODO.

**Comment by daveads** [comment]
@ezekg  Can you check it out now ?

**Review by ezekg — CHANGES_REQUESTED** [review]
One final change. Can you also run `go fmt` and squash all commits under a useful commit message?

File: cmd/new.go (hunk 4 above)
Comment (daveads) [reply cmd/new.go:66]: oh okayy

**Comment by daveads** [comment]
> One final change. Can you also run `go fmt` and squash all commits under a useful commit message?

alright

**Comment by daveads** [comment]
@ezekg  Done.

File: cmd/new.go (hunk 3 above)
Comment (ezekg) [inline cmd/new.go:61]: There are 2 TODOs here. Please keep the TODO about description. Only remove the TODO regarding metadata.

**Review by ezekg — CHANGES_REQUESTED** [review]

File: cmd/new.go (hunk 3 above)
Comment (daveads) [reply cmd/new.go:61]: > There are 2 TODOs here. Please keep the TODO about description. Only remove the TODO regarding metadata.

Done

**Review by ezekg — CHANGES_REQUESTED** [review]
Run `go fmt ./...`. Some of your files have not been formatted to Go standards.

Then squash.

**Comment by daveads** [comment]
> Run `go fmt ./...`. Some of your files have not been formatted to Go standards.
> 
> Then squash.

Done

**Comment by daveads** [comment]
> go fmt ./...



> Run `go fmt ./...`. Some of your files have not been formatted to Go standards.
> 
> Then squash.

@ezekg  CI is really necessary for this.

**Review by ezekg — APPROVED** [review]
